src/MOI_wrapper.jl

```diff
@@ -562,7 +562,7 @@ function MOI.get(
     return MOI.GreaterThan{Float64}(model.variable_info[c.value].lower_bound)
 end
 
-function MOI.get(model::Optimizer, ::MOI.ObjectiveFunction)
+function MOI.get(model::Optimizer, ::MOI.ObjectiveFunction{T})::T where {T}
     return model.objective
```

Comment:
```suggestion
    return convert(T, model.objective)
```

The annotation alone is pretty sneaky (I missed in the first read).

Reply:
I just copy-pasted the definition from `MOI/src/Utilities/model.jl`.

Reply:
Then we should probably also change `model.jl` haha.

Reply:
Agreed, this seems to aspect of the definition seems to be overlooked by many users so making it more visible is a good thing